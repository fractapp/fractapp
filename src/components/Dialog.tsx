import React from 'react';
import {StyleSheet, Text, View} from 'react-native';
import {BlueButton} from 'components/BlueButton';
import PopDialog, {DialogContent} from 'react-native-popup-dialog';

/**
 * Popup dialog component with 1 button
 * @category Components
 */
export const Dialog = ({
  title,
  text,
  visible,
  onPress,
}: {
  title: string;
  text: string;
  visible: boolean;
  onPress: () => void;
}) => {
  return (
    <PopDialog
      visible={visible}
      width={0.7}
      dialogTitle={<Text style={styles.title}>{title}</Text>}
      footer={
        <View style={styles.btnImg}>
          <BlueButton text={'OK'} height={40} onPress={onPress} />
        </View>
      }>
      <DialogContent style={styles.dialog}>
        <Text style={styles.description}>{text}</Text>
      </DialogContent>
    </PopDialog>
  );
};

const styles = StyleSheet.create({
  dialog: {
    alignItems: 'center',
    padding: 0,
  },
  title: {
    marginTop: 10,
    textAlign: 'center',
    fontSize: 17,
    fontFamily: 'Roboto-Regular',
    color: 'black',
  },
  description: {
    marginTop: 5,
    width: '100%',
    textAlign: 'center',
    fontSize: 14,
    fontFamily: 'Roboto-Regular',
    color: '#888888',
  },
  btnImg: {
    marginBottom: 12,
    width: '40%',
    alignSelf: 'center',
  },
});
