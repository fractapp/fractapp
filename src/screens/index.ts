export {ImportWallet} from './ImportWallet';
export {SaveSeed} from './SaveSeed';
export {SaveWallet} from './SaveWallet';
export {SettingWallet} from './SettingWallet';
export {Start} from './Start';
export {WalletDetails} from './WalletDetails';
export {WalletFileBackup} from './WalletFileBackup';
export {Wallets} from './Wallets';
export {Settings} from './Settings';
export {ConfirmSaveSeed} from './ConfirmSaveSeed';
export {ImportSeed} from './ImportSeed';
export {WalletFileImport} from './WalletFileImport';
export {GoogleDrivePicker} from './GoogleDrivePicker';
export {Receive} from './Receive';
export {Milestone} from './Milestone';
export {TransactionDetails} from './TransactionDetails';
export {SplashScreen} from './SplashScreen';
export {Backup} from './Backup';
export {NewPassCode} from './NewPassCode';
export {VerifyPassCode} from './VerifyPassCode';
export {EditProfile} from './EditProfile';
export {EditPhoneNumber} from './EditPhoneNumber';
export {SelectCountry} from './SelectCountry';
export {ConfirmCode} from './ConfirmCode';
export {EditEmail} from './EditEmail';
export {Navigation} from './Navigation';
export {Chats} from './Chats';
export {Chat} from './Chat';
export {Connecting} from './Connecting';
export {EditUsername} from './EditUsername';
export {EditName} from './EditName';
export {Search} from './Search';
export {SelectWallet} from './SelectWallet';
export {Send} from './Send';
export {EnterAmount} from './EnterAmount';
export {SendWithEnterAddress} from './SendWithEnterAddress';
