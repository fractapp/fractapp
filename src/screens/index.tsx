export { default as ImportWallet } from './ImportWallet';
export { default as SaveSeed } from './SaveSeed';

export { default as SaveWallet } from './SaveWallet';
export { default as SettingWallet } from './SettingWallet';
export { default as Start } from './Start';

export { default as WalletDetails } from './WalletDetails';

export { default as WalletEncryption } from './WalletEncryption';
export { default as Wallets } from './Wallets';
export { ConfirmSaveSeed } from './ConfirmSaveSeed'