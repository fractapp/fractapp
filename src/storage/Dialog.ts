import {createContext, Dispatch} from 'react';

/**
 * @namespace
 * @category Context storage
 */
namespace DialogStore {
  export enum Action {
    OPEN,
    CLOSE,
  }

  type State = {
    title: string;
    text: string;
    visible: boolean;
    onPress?: () => void;
  };

  export const initialState = (): State => ({
    text: '',
    title: '',
    visible: false,
  });

  export type ContextType = {
    state: State;
    dispatch: Dispatch<any>;
  };

  export const Context = createContext<ContextType>({
    state: initialState(),
    dispatch: () => null,
  });

  export function reducer(prevState: State, action: any): State {
    switch (action.type) {
      case Action.OPEN:
        return {
          title: action.title,
          text: action.text,
          onPress: action.onPress,
          visible: true,
        };
      case Action.CLOSE:
        let newDialog = Object.assign({}, prevState);
        newDialog.visible = false;
        return newDialog;
      default:
        return prevState;
    }
  }

  export const open = (title: string, text: string, onPress: () => void) => ({
    type: Action.OPEN,
    title: title,
    text: text,
    onPress: onPress,
  });
  export const close = () => ({
    type: Action.CLOSE,
  });
}
export default DialogStore;
