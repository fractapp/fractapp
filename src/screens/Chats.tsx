import React, {useEffect} from 'react';
import {
  FlatList,
  StyleSheet,
  TouchableHighlight,
  TouchableOpacity,
  View,
} from 'react-native';
import {ChatShortInfo} from 'components/ChatShortInfo';
import {ChatInfo} from 'types/chatInfo';
import Ionicons from 'react-native-vector-icons/Ionicons';
import { useDispatch, useSelector } from 'react-redux';
import ChatsStore from 'storage/Chats';
import UsersStore from 'storage/Users';
// @ts-ignore
import {MAIN_BOT_ID} from '@env';

/**
 * Chat list screen
 * @category Screens
 */
export const Chats = ({navigation}: {navigation: any}) => {
  const dispatch = useDispatch();
  const chatsState: ChatsStore.State = useSelector((state: any) => state.chats);
  const usersState: UsersStore.State = useSelector((state: any) => state.users);

  useEffect(() => {
    navigation.setOptions({
      headerRight: () => {
        return (
          <TouchableOpacity onPress={() => navigation.navigate('Search')}>
            <Ionicons name="add" size={32} color="#2AB2E2" />
          </TouchableOpacity>
        );
      },
    });
  }, []);

  const getChats = () => {
    const chats = Object.keys(chatsState.chatsInfo).map((key) => chatsState.chatsInfo[key])
      .filter((value) => value.lastMsgId != null)
      .sort((a, b) => {
          const msgOne = chatsState.chats[a.id]!.messages[a.lastMsgId!]!;
          const msgTwo = chatsState.chats[b.id]!.messages[b.lastMsgId!]!;
          if (msgOne.sender === MAIN_BOT_ID) {
            return -1;
          } else if (msgTwo.sender === MAIN_BOT_ID) {
            return 1;
          }

          return msgTwo.timestamp - msgOne.timestamp;
        }
      );

    return chats;
  };

  const renderItem = ({item}: {item: ChatInfo}) => {
    let user = usersState.users[item.id]!;

    if (!usersState.users[item.id]) {
      user = {
        isAddressOnly: false,
        title: 'Deleted',
        value: {
          id: item.id,
          name: 'Deleted',
          username: 'Deleted',
          avatarExt: '',
          lastUpdate: 0,
          addresses: null,
          isChatBot: false,
        },
      };
      dispatch(UsersStore.actions.setUsers([user]));
    }
    const msg = chatsState.chats[item.id]!.messages[item.lastMsgId!]!;

    return (
      <TouchableHighlight
        onPress={() => navigation.navigate('Chat', {chatId: item.id})}
        underlayColor="#f8f9fb">
        <ChatShortInfo
          notificationCount={item.notificationCount}
          message={msg.value}
          timestamp={msg.timestamp}
          user={user ?? {
            isAddressOnly: false,
            value: {
              id: '0',
              name: 'Deleted',
              username: '-',
              avatarExt: '',
              lastUpdate: 0,
              addresses: new Map(),
            },
          }}
        />
      </TouchableHighlight>
    );
  };

  return (
    <View style={styles.chats}>
      <FlatList
        ItemSeparatorComponent={() => <View style={styles.dividingLine} />}
        showsVerticalScrollIndicator={false}
        style={styles.list}
        data={getChats()}
        renderItem={renderItem}
        keyExtractor={(item, index) => String(index)}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  chats: {
    alignItems: 'center',
    justifyContent: 'center',
    flex: 1,
  },
  list: {
    flex: 1,
    width: '100%',
  },
  dividingLine: {
    marginLeft: 80,
    width: '100%',
    height: 1,
    backgroundColor: '#f5f5f5',
  },
});
