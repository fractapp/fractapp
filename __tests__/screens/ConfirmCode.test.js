import React, {useContext, useState} from 'react';
import renderer from 'react-test-renderer';
import {ConfirmCode} from 'screens/ConfirmCode';
import BackendApi from 'utils/api';
import GlobalStore from 'storage/Global';
import DialogStore from 'storage/Dialog';
import {fireEvent, render} from '@testing-library/react-native';
import Dialog from 'storage/Dialog';
import StringUtils from 'utils/string';

jest.mock('storage/DB', () => ({}));
jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useState: jest.fn(),
  useContext: jest.fn(),
  Linking: jest.fn(() => ({
    openURL: jest.fn(),
  })),
}));

jest.mock('@polkadot/keyring', () => {});
jest.mock('@polkadot/util', () => {});
jest.mock('@polkadot/keyring/types', () => {});
jest.mock('utils/api', () => ({
  auth: jest.fn(),
  sendCode: jest.fn(),
  CodeType: {
    Email: 12,
    Phone: 13,
  },
  CheckType: {
    Auth: 101,
  },
  CodeLength: 6,
}));
jest.mock('react-native-i18n', () => ({
  t: (value) => value,
}));

jest.useFakeTimers();
useState.mockImplementation((init) => [init, jest.fn()]);

const createContexts = () => {
  const dialogDispatch = jest.fn();
  const globalDispatch = jest.fn();
  useContext.mockReturnValueOnce({
    state: DialogStore.initialState(),
    dispatch: dialogDispatch,
  });
  useContext.mockReturnValueOnce({
    state: GlobalStore.initialState(),
    dispatch: globalDispatch,
  });

  return {
    dialogDispatch: dialogDispatch,
    globalDispatch: globalDispatch,
  };
};
const createStates = (code, locTime) => {
  const setCode = jest.fn();
  const setEditable = jest.fn();
  const setLockTime = jest.fn();
  const setBorderColor = jest.fn();

  useState
    .mockImplementationOnce((init) => [code, setCode])
    .mockImplementationOnce((init) => [init, setEditable])
    .mockImplementationOnce((init) => [locTime, setLockTime])
    .mockImplementationOnce((init) => [init, setBorderColor]);

  return {
    setCode: setCode,
    setEditable: setEditable,
    setLockTime: setLockTime,
    setBorderColor: setBorderColor,
  };
};

it('Test view', () => {
  createContexts();

  const tree = renderer
    .create(
      <ConfirmCode
        navigation={null}
        route={{
          params: {
            value: 'email',
            type: BackendApi.CodeType.Phone,
          },
        }}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Test with code (not full)', async () => {
  createContexts();

  const setters = createStates('1', 60);

  const navigate = await jest.fn();
  const tree = renderer
    .create(
      <ConfirmCode
        navigation={{navigate: navigate}}
        route={{
          params: {
            value: 'email',
            type: BackendApi.CodeType.Email,
          },
        }}
      />,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('Test with code (full / 200)', async () => {
  const dispatchers = createContexts();
  const setters = createStates('123123', 60);

  BackendApi.auth.mockReturnValueOnce(200);

  const navigate = jest.fn();
  const reset = jest.fn();
  const component = await render(
    <ConfirmCode
      navigation={{navigate: navigate, reset: reset}}
      route={{
        params: {
          value: 'email',
          type: BackendApi.CodeType.Email,
        },
      }}
    />,
  );

  expect(setters.setEditable).toBeCalledWith(false);
  expect(setters.setBorderColor).toBeCalledWith('#2AB2E2');
  expect(dispatchers.globalDispatch).toBeCalledWith(
    GlobalStore.setLoading(true),
  );
  expect(dispatchers.globalDispatch).toBeCalledWith(
    GlobalStore.signInFractapp(true),
  );
  expect(reset).toBeCalledWith({
    index: 1,
    actions: [navigate('Home'), navigate('EditProfile')],
  });
  expect(dispatchers.globalDispatch).toBeCalledWith(
    GlobalStore.setLoading(false),
  );
});

it('Test with code (throw)', async () => {
  const dispatchers = createContexts();
  const setters = createStates('123123', 60);

  BackendApi.auth.mockImplementationOnce(() => {
    throw new Error('error!');
  });

  const navigate = jest.fn();
  const reset = jest.fn();
  const component = await render(
    <ConfirmCode
      navigation={{navigate: navigate, reset: reset}}
      route={{
        params: {
          value: 'email',
          type: BackendApi.CodeType.Email,
        },
      }}
    />,
  );

  expect(setters.setEditable).toBeCalledWith(false);
  expect(setters.setBorderColor).toBeCalledWith('#2AB2E2');
  expect(dispatchers.globalDispatch).toBeCalledWith(
    GlobalStore.setLoading(true),
  );
  expect(setters.setCode).toBeCalledWith('');
  expect(setters.setBorderColor).toBeCalledWith('#EA4335');
  expect(setters.setEditable).toBeCalledWith(true);
  expect(dispatchers.globalDispatch).toBeCalledWith(
    GlobalStore.setLoading(false),
  );
});

it('Test with code (full / 400)', async () => {
  const dispatchers = createContexts();
  const setters = createStates('123123', 60);

  BackendApi.auth.mockReturnValueOnce(400);

  const navigate = jest.fn();
  const reset = jest.fn();
  const component = await render(
    <ConfirmCode
      navigation={{navigate: navigate, reset: reset}}
      route={{
        params: {
          value: 'email',
          type: BackendApi.CodeType.Email,
        },
      }}
    />,
  );

  expect(setters.setEditable).toBeCalledWith(false);
  expect(setters.setBorderColor).toBeCalledWith('#2AB2E2');
  expect(dispatchers.globalDispatch).toBeCalledWith(
    GlobalStore.setLoading(true),
  );
  expect(dispatchers.dialogDispatch.mock.calls[0][0]).toMatchSnapshot();
  await dispatchers.dialogDispatch.mock.calls[0][0].onPress();
  expect(dispatchers.dialogDispatch).toBeCalledWith(Dialog.close());

  expect(setters.setCode).toBeCalledWith('');
  expect(setters.setBorderColor).toBeCalledWith('#EA4335');

  expect(setters.setEditable).toBeCalledWith(true);
  expect(dispatchers.globalDispatch).toBeCalledWith(
    GlobalStore.setLoading(false),
  );
});

it('Test with code (full / 403)', async () => {
  const dispatchers = createContexts();
  const setters = createStates('123123', 60);

  BackendApi.auth.mockReturnValueOnce(403);

  const navigate = jest.fn();
  const reset = jest.fn();
  const goBack = jest.fn();
  const component = await render(
    <ConfirmCode
      navigation={{navigate: navigate, reset: reset, goBack}}
      route={{
        params: {
          value: 'email',
          type: BackendApi.CodeType.Email,
        },
      }}
    />,
  );

  expect(setters.setEditable).toBeCalledWith(false);
  expect(setters.setBorderColor).toBeCalledWith('#2AB2E2');
  expect(dispatchers.globalDispatch).toBeCalledWith(
    GlobalStore.setLoading(true),
  );
  expect(goBack).toBeCalled();
  expect(dispatchers.dialogDispatch.mock.calls[0][0]).toMatchSnapshot();
  await dispatchers.dialogDispatch.mock.calls[0][0].onPress();
  expect(dispatchers.dialogDispatch).toBeCalledWith(Dialog.close());

  expect(setters.setCode).toBeCalledWith('');
  expect(setters.setBorderColor).toBeCalledWith('#EA4335');

  expect(setters.setEditable).toBeCalledWith(true);
  expect(dispatchers.globalDispatch).toBeCalledWith(
    GlobalStore.setLoading(false),
  );
});

it('Test with code (full / 404)', async () => {
  const dispatchers = createContexts();
  const setters = createStates('123123', 60);

  BackendApi.auth.mockReturnValueOnce(404);

  const navigate = jest.fn();
  const reset = jest.fn();
  const goBack = jest.fn();
  const component = await render(
    <ConfirmCode
      navigation={{navigate: navigate, reset: reset, goBack}}
      route={{
        params: {
          value: 'email',
          type: BackendApi.CodeType.Email,
        },
      }}
    />,
  );

  expect(setters.setEditable).toBeCalledWith(false);
  expect(setters.setBorderColor).toBeCalledWith('#2AB2E2');
  expect(dispatchers.globalDispatch).toBeCalledWith(
    GlobalStore.setLoading(true),
  );
  expect(setters.setCode).toBeCalledWith('');
  expect(setters.setBorderColor).toBeCalledWith('#EA4335');

  expect(setters.setEditable).toBeCalledWith(true);
  expect(dispatchers.globalDispatch).toBeCalledWith(
    GlobalStore.setLoading(false),
  );
});

it('Test resend (200)', async () => {
  const dispatchers = createContexts();
  const setters = createStates('', 0);

  BackendApi.sendCode.mockReturnValueOnce(200);

  const navigate = jest.fn();
  const component = await render(
    <ConfirmCode
      navigation={{navigate: navigate}}
      route={{
        params: {
          value: 'email',
          type: BackendApi.CodeType.Email,
        },
      }}
    />,
  );
  await fireEvent.press(component.getByText(StringUtils.texts.ResendTitle));

  expect(setters.setLockTime).toBeCalledWith(60);
  expect(BackendApi.sendCode).toBeCalledWith(
    'email',
    BackendApi.CodeType.Email,
    BackendApi.CheckType.Auth,
  );
});

it('Test resend (400)', async () => {
  const dispatchers = createContexts();
  const setters = createStates('', 0);

  BackendApi.sendCode.mockReturnValueOnce(400);
  const navigate = jest.fn();
  const component = await render(
    <ConfirmCode
      navigation={{navigate: navigate}}
      route={{
        params: {
          value: 'email',
          type: BackendApi.CodeType.Email,
        },
      }}
    />,
  );
  await fireEvent.press(component.getByText(StringUtils.texts.ResendTitle));

  expect(setters.setLockTime).toBeCalledWith(60);
  expect(BackendApi.sendCode).toBeCalledWith(
    'email',
    BackendApi.CodeType.Email,
    BackendApi.CheckType.Auth,
  );
  expect(dispatchers.dialogDispatch.mock.calls[0][0]).toMatchSnapshot();
  await dispatchers.dialogDispatch.mock.calls[0][0].onPress();
  expect(dispatchers.dialogDispatch).toBeCalledWith(Dialog.close());
});

it('Test resend (404/email)', async () => {
  const dispatchers = createContexts();
  const setters = createStates('', 0);

  BackendApi.sendCode.mockReturnValueOnce(404);
  const navigate = jest.fn();
  const component = await render(
    <ConfirmCode
      navigation={{navigate: navigate}}
      route={{
        params: {
          value: 'email',
          type: BackendApi.CodeType.Email,
        },
      }}
    />,
  );
  await fireEvent.press(component.getByText(StringUtils.texts.ResendTitle));

  expect(setters.setLockTime).toBeCalledWith(60);
  expect(BackendApi.sendCode).toBeCalledWith(
    'email',
    BackendApi.CodeType.Email,
    BackendApi.CheckType.Auth,
  );
  expect(dispatchers.dialogDispatch.mock.calls[0][0]).toMatchSnapshot();
  await dispatchers.dialogDispatch.mock.calls[0][0].onPress();
  expect(dispatchers.dialogDispatch).toBeCalledWith(Dialog.close());
});

it('Test resend (404/phone)', async () => {
  const dispatchers = createContexts();
  const setters = createStates('', 0);

  BackendApi.sendCode.mockReturnValueOnce(404);
  const navigate = jest.fn();
  const component = await render(
    <ConfirmCode
      navigation={{navigate: navigate}}
      route={{
        params: {
          value: 'phone',
          type: BackendApi.CodeType.Phone,
        },
      }}
    />,
  );
  await fireEvent.press(component.getByText(StringUtils.texts.ResendTitle));

  expect(setters.setLockTime).toBeCalledWith(60);
  expect(BackendApi.sendCode).toBeCalledWith(
    'phone',
    BackendApi.CodeType.Phone,
    BackendApi.CheckType.Auth,
  );
  expect(dispatchers.dialogDispatch.mock.calls[0][0]).toMatchSnapshot();
  await dispatchers.dialogDispatch.mock.calls[0][0].onPress();
  expect(dispatchers.dialogDispatch).toBeCalledWith(Dialog.close());
});
