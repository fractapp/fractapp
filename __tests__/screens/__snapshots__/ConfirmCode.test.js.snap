// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test resend (400) 1`] = `
Object {
  "onPress": [Function],
  "text": "",
  "title": "[missing \\"en.service_unavailable_title\\" translation]",
  "type": 0,
}
`;

exports[`Test resend (404/email) 1`] = `
Object {
  "onPress": [Function],
  "text": "[missing \\"en.invalid_email_text\\" translation]",
  "title": "[missing \\"en.invalid_email_title\\" translation]",
  "type": 0,
}
`;

exports[`Test resend (404/phone) 1`] = `
Object {
  "onPress": [Function],
  "text": "[missing \\"en.invalid_phone_number_text\\" translation]",
  "title": "[missing \\"en.invalid_phone_number_title\\" translation]",
  "type": 0,
}
`;

exports[`Test view 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "paddingTop": 10,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#888888",
        "fontFamily": "Roboto-Regular",
        "fontSize": 16,
        "fontStyle": "normal",
        "fontWeight": "normal",
        "textAlign": "center",
        "width": "80%",
      }
    }
  >
    [missing "en.confirmation_code_phone_number_text" translation]
  </Text>
  <Text
    style={
      Object {
        "color": "#067BCD",
        "fontFamily": "Roboto-Medium",
        "fontSize": 17,
        "fontStyle": "normal",
        "fontWeight": "normal",
        "marginBottom": 30,
      }
    }
  >
    +
    email
  </Text>
  <View
    style={
      Object {
        "alignContent": "center",
        "flexDirection": "row",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      blurOnSubmit={false}
      caretHidden={true}
      keyboardType="phone-pad"
      maxLength={1}
      onChangeText={[Function]}
      onFocus={[Function]}
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "borderColor": "#CCCCCC",
            "borderRadius": 13,
            "borderWidth": 1,
            "color": "#2AB2E2",
            "fontFamily": "Roboto-Regular",
            "fontSize": 23,
            "fontStyle": "normal",
            "fontWeight": "normal",
            "height": 50,
            "marginLeft": 5,
            "marginRight": 5,
            "textAlign": "center",
            "width": 50,
          },
          Object {
            "borderColor": "#CCCCCC",
          },
        ]
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <TextInput
      allowFontScaling={true}
      blurOnSubmit={false}
      caretHidden={true}
      editable={true}
      keyboardType="phone-pad"
      maxLength={1}
      onChangeText={[Function]}
      onFocus={[Function]}
      onKeyPress={[Function]}
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "borderColor": "#CCCCCC",
            "borderRadius": 13,
            "borderWidth": 1,
            "color": "#2AB2E2",
            "fontFamily": "Roboto-Regular",
            "fontSize": 23,
            "fontStyle": "normal",
            "fontWeight": "normal",
            "height": 50,
            "marginLeft": 5,
            "marginRight": 5,
            "textAlign": "center",
            "width": 50,
          },
          Object {
            "borderColor": "#CCCCCC",
          },
        ]
      }
      underlineColorAndroid="transparent"
    />
    <TextInput
      allowFontScaling={true}
      blurOnSubmit={false}
      caretHidden={true}
      editable={true}
      keyboardType="phone-pad"
      maxLength={1}
      onChangeText={[Function]}
      onFocus={[Function]}
      onKeyPress={[Function]}
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "borderColor": "#CCCCCC",
            "borderRadius": 13,
            "borderWidth": 1,
            "color": "#2AB2E2",
            "fontFamily": "Roboto-Regular",
            "fontSize": 23,
            "fontStyle": "normal",
            "fontWeight": "normal",
            "height": 50,
            "marginLeft": 5,
            "marginRight": 5,
            "textAlign": "center",
            "width": 50,
          },
          Object {
            "borderColor": "#CCCCCC",
          },
        ]
      }
      underlineColorAndroid="transparent"
    />
    <TextInput
      allowFontScaling={true}
      blurOnSubmit={false}
      caretHidden={true}
      editable={true}
      keyboardType="phone-pad"
      maxLength={1}
      onChangeText={[Function]}
      onFocus={[Function]}
      onKeyPress={[Function]}
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "borderColor": "#CCCCCC",
            "borderRadius": 13,
            "borderWidth": 1,
            "color": "#2AB2E2",
            "fontFamily": "Roboto-Regular",
            "fontSize": 23,
            "fontStyle": "normal",
            "fontWeight": "normal",
            "height": 50,
            "marginLeft": 5,
            "marginRight": 5,
            "textAlign": "center",
            "width": 50,
          },
          Object {
            "borderColor": "#CCCCCC",
          },
        ]
      }
      underlineColorAndroid="transparent"
    />
    <TextInput
      allowFontScaling={true}
      blurOnSubmit={false}
      caretHidden={true}
      editable={true}
      keyboardType="phone-pad"
      maxLength={1}
      onChangeText={[Function]}
      onFocus={[Function]}
      onKeyPress={[Function]}
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "borderColor": "#CCCCCC",
            "borderRadius": 13,
            "borderWidth": 1,
            "color": "#2AB2E2",
            "fontFamily": "Roboto-Regular",
            "fontSize": 23,
            "fontStyle": "normal",
            "fontWeight": "normal",
            "height": 50,
            "marginLeft": 5,
            "marginRight": 5,
            "textAlign": "center",
            "width": 50,
          },
          Object {
            "borderColor": "#CCCCCC",
          },
        ]
      }
      underlineColorAndroid="transparent"
    />
    <TextInput
      allowFontScaling={true}
      blurOnSubmit={false}
      caretHidden={true}
      editable={true}
      keyboardType="phone-pad"
      maxLength={1}
      onChangeText={[Function]}
      onFocus={[Function]}
      onKeyPress={[Function]}
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "borderColor": "#CCCCCC",
            "borderRadius": 13,
            "borderWidth": 1,
            "color": "#2AB2E2",
            "fontFamily": "Roboto-Regular",
            "fontSize": 23,
            "fontStyle": "normal",
            "fontWeight": "normal",
            "height": 50,
            "marginLeft": 5,
            "marginRight": 5,
            "textAlign": "center",
            "width": 50,
          },
          Object {
            "borderColor": "#CCCCCC",
          },
        ]
      }
      underlineColorAndroid="transparent"
    />
  </View>
  <Text
    style={
      Object {
        "color": "#888888",
        "fontFamily": "Roboto-Regular",
        "fontSize": 16,
        "fontStyle": "normal",
        "fontWeight": "normal",
        "marginTop": 30,
      }
    }
  >
    [missing "en.resend_text" translation] 1:00
  </Text>
</View>
`;

exports[`Test with code (full / 400) 1`] = `
Object {
  "onPress": [Function],
  "text": "",
  "title": "[missing \\"en.service_unavailable_title\\" translation]",
  "type": 0,
}
`;

exports[`Test with code (full / 403) 1`] = `
Object {
  "onPress": [Function],
  "text": "",
  "title": "[missing \\"en.different_address_title\\" translation]",
  "type": 0,
}
`;

exports[`Test with code (not full) 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "paddingTop": 10,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#888888",
        "fontFamily": "Roboto-Regular",
        "fontSize": 16,
        "fontStyle": "normal",
        "fontWeight": "normal",
        "textAlign": "center",
        "width": "80%",
      }
    }
  >
    [missing "en.confirmation_code_email_text" translation]
  </Text>
  <Text
    style={
      Object {
        "color": "#067BCD",
        "fontFamily": "Roboto-Medium",
        "fontSize": 17,
        "fontStyle": "normal",
        "fontWeight": "normal",
        "marginBottom": 30,
      }
    }
  >
    
    email
  </Text>
  <View
    style={
      Object {
        "alignContent": "center",
        "flexDirection": "row",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      blurOnSubmit={false}
      caretHidden={true}
      keyboardType="phone-pad"
      maxLength={1}
      onChangeText={[Function]}
      onFocus={[Function]}
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "borderColor": "#CCCCCC",
            "borderRadius": 13,
            "borderWidth": 1,
            "color": "#2AB2E2",
            "fontFamily": "Roboto-Regular",
            "fontSize": 23,
            "fontStyle": "normal",
            "fontWeight": "normal",
            "height": 50,
            "marginLeft": 5,
            "marginRight": 5,
            "textAlign": "center",
            "width": 50,
          },
          Object {
            "borderColor": "#CCCCCC",
          },
        ]
      }
      underlineColorAndroid="transparent"
      value="1"
    />
    <TextInput
      allowFontScaling={true}
      blurOnSubmit={false}
      caretHidden={true}
      editable={true}
      keyboardType="phone-pad"
      maxLength={1}
      onChangeText={[Function]}
      onFocus={[Function]}
      onKeyPress={[Function]}
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "borderColor": "#CCCCCC",
            "borderRadius": 13,
            "borderWidth": 1,
            "color": "#2AB2E2",
            "fontFamily": "Roboto-Regular",
            "fontSize": 23,
            "fontStyle": "normal",
            "fontWeight": "normal",
            "height": 50,
            "marginLeft": 5,
            "marginRight": 5,
            "textAlign": "center",
            "width": 50,
          },
          Object {
            "borderColor": "#CCCCCC",
          },
        ]
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <TextInput
      allowFontScaling={true}
      blurOnSubmit={false}
      caretHidden={true}
      editable={true}
      keyboardType="phone-pad"
      maxLength={1}
      onChangeText={[Function]}
      onFocus={[Function]}
      onKeyPress={[Function]}
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "borderColor": "#CCCCCC",
            "borderRadius": 13,
            "borderWidth": 1,
            "color": "#2AB2E2",
            "fontFamily": "Roboto-Regular",
            "fontSize": 23,
            "fontStyle": "normal",
            "fontWeight": "normal",
            "height": 50,
            "marginLeft": 5,
            "marginRight": 5,
            "textAlign": "center",
            "width": 50,
          },
          Object {
            "borderColor": "#CCCCCC",
          },
        ]
      }
      underlineColorAndroid="transparent"
    />
    <TextInput
      allowFontScaling={true}
      blurOnSubmit={false}
      caretHidden={true}
      editable={true}
      keyboardType="phone-pad"
      maxLength={1}
      onChangeText={[Function]}
      onFocus={[Function]}
      onKeyPress={[Function]}
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "borderColor": "#CCCCCC",
            "borderRadius": 13,
            "borderWidth": 1,
            "color": "#2AB2E2",
            "fontFamily": "Roboto-Regular",
            "fontSize": 23,
            "fontStyle": "normal",
            "fontWeight": "normal",
            "height": 50,
            "marginLeft": 5,
            "marginRight": 5,
            "textAlign": "center",
            "width": 50,
          },
          Object {
            "borderColor": "#CCCCCC",
          },
        ]
      }
      underlineColorAndroid="transparent"
    />
    <TextInput
      allowFontScaling={true}
      blurOnSubmit={false}
      caretHidden={true}
      editable={true}
      keyboardType="phone-pad"
      maxLength={1}
      onChangeText={[Function]}
      onFocus={[Function]}
      onKeyPress={[Function]}
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "borderColor": "#CCCCCC",
            "borderRadius": 13,
            "borderWidth": 1,
            "color": "#2AB2E2",
            "fontFamily": "Roboto-Regular",
            "fontSize": 23,
            "fontStyle": "normal",
            "fontWeight": "normal",
            "height": 50,
            "marginLeft": 5,
            "marginRight": 5,
            "textAlign": "center",
            "width": 50,
          },
          Object {
            "borderColor": "#CCCCCC",
          },
        ]
      }
      underlineColorAndroid="transparent"
    />
    <TextInput
      allowFontScaling={true}
      blurOnSubmit={false}
      caretHidden={true}
      editable={true}
      keyboardType="phone-pad"
      maxLength={1}
      onChangeText={[Function]}
      onFocus={[Function]}
      onKeyPress={[Function]}
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "borderColor": "#CCCCCC",
            "borderRadius": 13,
            "borderWidth": 1,
            "color": "#2AB2E2",
            "fontFamily": "Roboto-Regular",
            "fontSize": 23,
            "fontStyle": "normal",
            "fontWeight": "normal",
            "height": 50,
            "marginLeft": 5,
            "marginRight": 5,
            "textAlign": "center",
            "width": 50,
          },
          Object {
            "borderColor": "#CCCCCC",
          },
        ]
      }
      underlineColorAndroid="transparent"
    />
  </View>
  <Text
    style={
      Object {
        "color": "#888888",
        "fontFamily": "Roboto-Regular",
        "fontSize": 16,
        "fontStyle": "normal",
        "fontWeight": "normal",
        "marginTop": 30,
      }
    }
  >
    [missing "en.resend_text" translation] 1:00
  </Text>
</View>
`;
